src/App.tsx

```diff
@@ -10,18 +13,57 @@ import TableContainer from '@mui/material/TableContainer';
 import TableRow from '@mui/material/TableRow';
 import Paper from '@mui/material/Paper';
 import { TableHead } from "@mui/material";
+import { log_levels } from "./schema/Log_Levels.ts";
+
+const apiURL = "/api/views/search/sync";
+const password = "token";
+let username: string
+const query: Query_String = {};
+
+const ACTIONS = {
+  LOGFILTER: "level",
+  BEAMLINE: "beamline",
+  APP: "application",
+};
+
 
 type getMessageReturn = [string[],string[],string[],string[],number[],string[]]
 
 function App() {
+  // API responses to be shown
   const [time, setTime] = useState<string[]>([]);
   const [host, setHost] = useState<string[]>([]);
   const [debug, setDebug] = useState<string[]>([]);
   const [logs, setLogs] = useState<string[]>([]);
   const [log_lvl, setLog_lvl] = useState<number[]>([]);
   const [app_name,setApp_name] = useState<string[]>([]);
+  const [logfilter, setLogfilter] = useState<number>(7)
+  // Log_Menu Props
+  const handleLogFilterChange = (newLogFilterValue: number) => {
+    setLogfilter(newLogFilterValue);
+    handlePayload({type: ACTIONS.LOGFILTER, log_level: newLogFilterValue});
+  }
 
-  useEffect(() => {
+  function reducer(payload:Payload_interface,action:action_type) {
```

Comment:
I'm curious about why you chose to have a single reducer and a switch statement to cover 3 fields rather than have 3 reducers. I'm not really sure of the relative pros and cons or accepted best practice. Also interested to know if @abigailalexander has an opinion.

Reply:
The reason why I wanted to use one reducer was that the only thing changing was the Payload variable, but in different ways depending on what is changed. Also thinking about in the future, any other queries that might be needed e.g. changing a timestamp or other filters, can just add another case in the switch statement.

Reply:
Using a reducer with a switch statement is fine, especially if you're performing varying actions on the same payload @callumforrester. I think you'd probably need multiple reducers if you were dealing with very different parts of the state, but one is enough for just modifying the payload

Reply:
Great, thanks both!